src/main/java/com/wasilewskyy/github_proxy/service/GithubRepositoryService.java

```diff
@@ -0,0 +1,116 @@
+package com.wasilewskyy.github_proxy.service;
+
+import com.wasilewskyy.github_proxy.client.GithubClient;
+import com.wasilewskyy.github_proxy.exception.RepositoryNotFoundException;
+import com.wasilewskyy.github_proxy.mapper.GithubRepositoryMapper;
+import com.wasilewskyy.github_proxy.model.GithubRepository;
+import com.wasilewskyy.github_proxy.model.GithubRepositoryDTO;
+import com.wasilewskyy.github_proxy.model.GithubRepositoryResponse;
+import com.wasilewskyy.github_proxy.repository.GithubRepositoryJPA;
+import feign.FeignException;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class GithubRepositoryService {
+
+    private final GithubRepositoryJPA repositoryJPA;
+    private final GithubRepositoryMapper githubRepositoryMapper;
+    private final GithubClient githubClient;
+
+    public GithubRepositoryDTO getRepositoryFromGithub(String owner, String repositoryName) {
+        GithubRepositoryResponse gitHubApiResponse;
+        try {
```

Comment:
try {
            gitHubApiResponse = githubClient.getRepository(owner, repositoryName);
            return githubRepositoryMapper.fromGithubResponse(gitHubApiResponse);
        } catch (FeignException.NotFound ex) {
            throw new RepositoryNotFoundException("Requested repository could not be found.");
        }